blur_no-alt.php

```diff
@@ -54,70 +86,169 @@ public function blur_no_alt_message_display_create_admin_page() {
+			$sanitized['show_or_hide_0'] = 'Show' === $input['show_or_hide_0'] ? 'Show' : false;
+		}
+
+		if ( isset( $input['blur_on_front'] ) ) {
+			$sanitized['blur_on_front'] = intval( $input['blur_on_front'] );
 		}
 
 		return $sanitized;
 	}
 
-	public function blur_no_alt_message_display_section_info() {
-		
+	/**
+	 * Output help text about front-end blur capabilities filter
+	 */
+	public function blur_no_alt_front_end_section_callback() {
+		echo '<p>' . wp_kses(
+				__( 'When enabled, this option applies to all users with <code>edit_pages</code> capability. Customize capability with <code>blur_no_alt_front_end_blur_capability</code> filter.' ),
+				array( 'code' => array() )
+			) . '</p>';
 	}
 
-	public function show_or_hide_0_callback() {
-		?> <fieldset><?php $checked = ( isset( $this->blur_no_alt_message_display_options['show_or_hide_0'] ) && $this->blur_no_alt_message_display_options['show_or_hide_0'] === 'Show' ) ? 'checked' : '' ; ?>
-		<label for="show_or_hide_0-0"><input type="radio" name="blur_no_alt_message_display_option_name[show_or_hide_0]" id="show_or_hide_0-0" value="Show" <?php echo $checked; ?>> <?php esc_html_e( 'Show', 'blur-no-alt' ); ?></label><br>
-		<?php $checked = ( isset( $this->blur_no_alt_message_display_options['show_or_hide_0'] ) && $this->blur_no_alt_message_display_options['show_or_hide_0'] === 'Hide' ) ? 'checked' : '' ; ?>
-		<label for="show_or_hide_0-1"><input type="radio" name="blur_no_alt_message_display_option_name[show_or_hide_0]" id="show_or_hide_0-1" value="Hide" <?php echo $checked; ?>> <?php esc_html_e( 'Hide', 'blur-no-alt' ); ?></label></fieldset> <?php
+	/**
+	 * Outputs radio buttons for
+	 */
+	public function show_or_hide_setting_callback() {
+		$show_is_checked = (
+			isset( $this->blur_no_alt_message_display_options['show_or_hide_0'] ) &&
+			'Show' === $this->blur_no_alt_message_display_options['show_or_hide_0']
+		);
+		?>
+		<fieldset>
+			<input type="checkbox" name="blur_no_alt_message_display_option_name[show_or_hide_0]" id="show_or_hide_0-0" value="Show" <?php echo $show_is_checked ? 'checked' : ''; ?>>
+			<label for="show_or_hide_0-0">
+				<?php esc_html_e( 'Show informational message to editors', 'blur-no-alt' ); ?>
+			</label>
+		</fieldset>
+		<?php
+	}
+
+	/**
+	 * Outputs checkbox for blur_on_front setting
+	 */
+	public function blur_no_alt_on_front_setting_callback() {
+		$checked = (
+			isset( $this->blur_no_alt_message_display_options['blur_on_front'] ) &&
+			1 === $this->blur_no_alt_message_display_options['blur_on_front']
+		) ? 'checked' : '';
+		?>
+		<input type="checkbox" name="blur_no_alt_message_display_option_name[blur_on_front]" id="blur_on_front" value="1" <?php echo esc_attr( $checked ); ?>>
+		<label for="blur_on_front">
+			<?php esc_html_e( 'Blur images on front-end for logged-in users', 'blur-no-alt' ); ?>
+		</label>
+		<?php
```

Comment:
I'd prefer radio buttons with these. [This is probably more rationale than you want about it](https://www.sarasoueidan.com/blog/one-checkbox-or-two-radio-buttons/).

Reply:
Thanks again for sharing that link. I had read that article a while ago, and was very happy to revisit it. Interestingly, after reading it, I didn't conclude that there was a cut-and-dry right answer in either direction. If anything, I'd say I'd still lean checkboxes just because:

- I have a vague guess that unchecked is the default a majority of folks will want.
- A very good point from the article: Unchecked can mean both A) intentionally left unchecked or B) not yet reviewed. Since the plugin adds two new options that are both off-by-default and can only be reviewed after install, that feels right to me.

Bigger picture, related to the first list item in my previous comment, the "Show" and "Hide" labels would need additional context for screen readers to be clearly understood, but the settings page already felt pretty busy and verbose. I had a hard time writing content for the table cell "labels" that didn't feel redundant with the checkbox labels.

So I'm not sure where that leaves us. I think preference is to leave these changes alone on account of the ambiguity and follow the general pattern that WP establishes on its own default settings pages. If you're only willing to merge this if I change both to radios, I'll work to switch it over (probably no earlier than next week).